tests/unit/io/ARFFFile_unittest.cc

```diff
@@ -0,0 +1,180 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#include <shogun/io/ARFFFile.h>
+
+#include <gtest/gtest.h>
+#include <shogun/features/DenseFeatures.h>
+
+using namespace shogun;
+
+TEST(ARFFFileTest, Parse_numeric)
+{
+	std::string test = "@relation test_numeric \n"
+	                   "%\n"
+	                   "% \n"
+	                   "@attribute VAR1 numeric \n"
+	                   "@attribute VAR2 real \n"
+	                   "% \n"
+	                   "% \n"
+	                   "@data \n"
+	                   "50, 5.1 \n"
+	                   "45, 4.13 ";
+	auto ss = std::make_shared<std::istringstream>(test);
+	auto s = std::shared_ptr<std::istream>(ss);
+
+	auto parser = std::make_unique<ARFFDeserializer>(s);
+	parser->read();
+	auto result = parser->get_features();
+	ASSERT_EQ(result->get_num_feature_obj(), 2);
+
+	auto col1 =
+	    dynamic_cast<CDenseFeatures<float64_t>*>(result->get_feature_obj(0));
+	auto col2 =
+	    dynamic_cast<CDenseFeatures<float64_t>*>(result->get_feature_obj(1));
```

Comment:
maybe `->as()` since otherwise you need to nullptr check?

Reply:
yup